### PR TITLE
[Storage] `az storage blob set-tier`: Migrate to Track 2 to support setting rehydrate priority

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Feature request: #13840

**Description**  
Migrate to Track 2 to support setting rehydrate priority

**Testing Guide**  
Run `az storage blob set-tier -n blob_name -c container_name --tier Cool -r High`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S171

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -495,11 +495,18 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+                                                                   blob_rehydrate_priority_validator)
 
+        c.argument('container_name', container_name_type)
+        c.argument('blob_name', options_list=('--name', '-n'), help="The blob name")
         c.argument('blob_type', options_list=('--type', '-t'), arg_type=get_enum_type(('block', 'page')))
         c.argument('tier', validator=blob_tier_validator)
         c.argument('timeout', type=int)
+        c.argument('rehydrate_priority', options_list=('--rehydrate-priority', '-r'),
+                   arg_type=get_enum_type(('High', 'Standard')), validator=blob_rehydrate_priority_validator,
+                   help="Indicates the priority with which to rehydrate an archived blob. "
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:508]:
```suggestion
                   arg_type=get_enum_type(('High', 'Standard')), validator=blob_rehydrate_priority_validator, is_preview=True,
                   help="Indicate the priority with which to rehydrate an archived blob. "
```

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 2]
```diff
@@ -1077,6 +1077,15 @@ def blob_tier_validator(cmd, namespace):
         raise ValueError('Blob tier is only applicable to block or page blob.')
 
 
+def blob_rehydrate_priority_validator(namespace):
+    if namespace.blob_type == 'page' and namespace.rehydrate_priority:
+        raise ValueError('--rehydrate-priority is only applicable to block blob.')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py:1082]: What about append blob?

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 3]
```diff
@@ -1077,6 +1077,15 @@ def blob_tier_validator(cmd, namespace):
...
+    if namespace.tier == 'Archive' and namespace.rehydrate_priority:
+        raise ValueError('--rehydrate-priority is only applicable to rehydrate blob data from the archive tier.')
+    if namespace.rehydrate_priority is None:
+        namespace.rehydrate_priority = 'Standard'
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py:1086]: According to the design here, you will provide default value for rehydrate_priority, right? Is it expected?

File: src/azure-cli/azure/cli/command_modules/storage/operations/blob.py [hunk 4]
```diff
@@ -632,3 +632,12 @@ def show_blob_v2(cmd, client, container_name, blob_name, snapshot=None, lease_id
...
     return blob
+
+
+def set_blob_tier_v2(client, container_name, blob_name, tier, blob_type='block', rehydrate_priority=None, timeout=None):
+    if blob_type == 'block':
+        return client.set_standard_blob_tier(standard_blob_tier=tier, rehydrate_priority=rehydrate_priority,
+                                             timeout=timeout)
+    if blob_type == 'page':
+        return client.set_premium_page_blob_tier(premium_page_blob_tier=tier, timeout=timeout)
+    raise ValueError('Blob tier is only applicable to block or page blob.')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/storage/operations/blob.py:643]: what about append blob?

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py (hunk 2 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_validators.py:1082]: oh. It will be rejected. never mind

**Comment by Juliehzl** [comment]
approved with several comments

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py (hunk 3 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/storage/_validators.py:1086]: Yes, that's how it's designed

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 5]
```diff
@@ -495,11 +495,18 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+                                                                   blob_rehydrate_priority_validator)
 
+        c.argument('container_name', container_name_type)
+        c.argument('blob_name', options_list=('--name', '-n'), help="The blob name")
         c.argument('blob_type', options_list=('--type', '-t'), arg_type=get_enum_type(('block', 'page')))
         c.argument('tier', validator=blob_tier_validator)
         c.argument('timeout', type=int)
+        c.argument('rehydrate_priority', options_list=('--rehydrate-priority', '-r'),
+                   arg_type=get_enum_type(('High', 'Standard')), validator=blob_rehydrate_priority_validator, is_preview=True,
+                   help="Indicate the priority with which to rehydrate an archived blob. "
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:508]:
```suggestion
                   arg_type=get_enum_type(('High', 'Standard')), validator=blob_rehydrate_priority_validator, 
                   is_preview=True, help="Indicate the priority with which to rehydrate an archived blob. "
```

**Pull request merged**